src/main/java/com/diablo3CharViewer/json_mappers/ItemMapper.java

```diff
@@ -0,0 +1,97 @@
+package com.diablo3CharViewer.json_mappers;
+
+import com.diablo3CharViewer.api_handlers.ItemHandlerApi;
+import com.diablo3CharViewer.data_models.ItemArmorDataModel;
+import com.diablo3CharViewer.data_models.ItemDataModel;
+import com.diablo3CharViewer.data_models.ItemWeaponDataModel;
+import com.diablo3CharViewer.token.FetchToken;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ItemMapper extends ItemHandlerApi {
```

Comment:
Why does ItemMapper fetch items from API? This is very confusing. I think you are not using inheritance properly. ItemMapper is not ItemHandlerAPI, it uses it.

Reply:
Good point. It is better to avoid inheritance for this class and make method genrateRequest() static instead because it is only used once in any Mapper class.